### PR TITLE
Added fixes for GBP workflow

### PR DESCRIPTION
**Pull request by tbachman**
This adds fixes needed to support GBP workflow. It also
adds some extra error logging and fixes some tests.

Signed-off-by: Thomas Bachman tbachman@yahoo.com

### CODE REVIEW

File: networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py [hunk 1]
```diff
@@ -161,19 +189,32 @@ def apic_driver(self):
...
                 self._get_ext_net_name = self._get_ext_net_name_gbp
+                self._get_vrf_context = self._get_vrf_context_gbp
+                self._get_vrf_details = self._get_vrf_details_gbp
+            except KeyError:
+                    LOG.info(_("GBP service plugin not present -- skipping "
+                               "dynamic load of GBP policy driver."))
             except Exception:
-                pass
+                    LOG.error(_("GBP service plugin present, but dynamic "
+                                "load of GBP policy driver failed."))
```
Comment (kentwu1) [inline networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py]: should we just return right here?

File: networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py [hunk 2]
```diff
@@ -237,16 +278,18 @@ def allocate_hosting_port(self, context, router_id, port_db, network_type,
         networks = self._core_plugin.get_networks(
             context.elevated(), {'id': [network_id]})
         l3out_network = networks[0]
+        l3out_name = self.get_ext_net_name(l3out_network['name'])
```
Comment (kentwu1) [inline networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py:272]: this line is redundant as line#290 also does the same

File: networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py [hunk 3]
```diff
@@ -161,19 +189,32 @@ def apic_driver(self):
...
                             'apic'].obj)
-                self._get_vrf_context = self._get_vrf_context_gbp
                 self._get_ext_net_name = self._get_ext_net_name_gbp
+                self._get_vrf_context = self._get_vrf_context_gbp
+                self._get_vrf_details = self._get_vrf_details_gbp
+            except KeyError:
+                    LOG.info(_("GBP service plugin not present -- skipping "
+                               "dynamic load of GBP policy driver."))
             except Exception:
-                pass
```
Comment (kentwu1) [inline networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py:167]: how to hit this except condition? I don't see any load operation in this call:

manager.NeutronManager.get_service_plugins()[
                         'GROUP_POLICY'].policy_driver_manager.policy_drivers[
                             'apic'].obj

same thing for neutron case.

File: networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py (hunk 1 above)
Comment (tbachman) [inline networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py]: Actually, I think it's wrong in another way -- I don't think we should be expecting anything other than a KeyError, so I probably should get rid of the second exception. Will fix.

File: networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py (hunk 2 above)
Comment (tbachman) [inline networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py:272]: Good catch -- will fix.

File: networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py (hunk 3 above)
Comment (tbachman) [inline networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py:167]: Yeah -- I should just get rid of the second except. Will fix.

**Comment by tbachman** [comment]
Addressed reviewer comments.

**Pull request merged**